src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -34,14 +35,18 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(Authentication authentication) {
+        User user = userService.findByEmail(authentication.getName()).orElseThrow(() ->
+                new RuntimeException("Can't find shopping cart for this user"));
```

Comment:
```suggestion
                new RuntimeException("Can't user with email: " + authentication.getName()));
```
the same for other similar cases